src/main/java/org/broadinstitute/hellbender/utils/read/AlignmentUtils.java

```diff
@@ -200,48 +199,57 @@ public static GATKRead createReadAlignedToRef(final GATKRead originalRead,
         return Arrays.copyOfRange(bases, basesStart, basesStop + 1);
     }
 
-    public static byte[] getBasesAlignedOneToOne(final GATKRead read) {
-        return getSequenceAlignedOneToOne(read, r -> r.getBasesNoCopy(), GAP_CHARACTER);
+    public static Tuple<byte[], byte[]> getBasesAndBaseQualitiesAlignedOneToOne(final GATKRead read) {
+        return getBasesAndBaseQualitiesAlignedOneToOne(read, GAP_CHARACTER, (byte)0);
     }
 
-    public static byte[] getBaseQualsAlignedOneToOne(final GATKRead read) {
-        return getSequenceAlignedOneToOne(read, r -> r.getBaseQualitiesNoCopy(), (byte)0);
-    }
-
-    public static byte[] getSequenceAlignedOneToOne(final GATKRead read, final Function<GATKRead, byte[]> bytesProvider, final byte padWith) {
+    public static Tuple<byte[], byte[]> getBasesAndBaseQualitiesAlignedOneToOne(final GATKRead read, final byte gapCharacter, final byte qualityPadCharacter) {
         Utils.nonNull(read);
-        Utils.nonNull(bytesProvider);
-        final Cigar cigar = read.getCigar();
-        final byte[] sequence = bytesProvider.apply(read);
-
-        if (!cigar.containsOperator(CigarOperator.DELETION) && !cigar.containsOperator(CigarOperator.INSERTION)) {
-            return sequence;
+        final byte[] bases = read.getBasesNoCopy();
+        final byte[] baseQualities = read.getBaseQualitiesNoCopy();
+        final int numCigarElements = read.numCigarElements();
+        boolean sawIndel = false;
+
+        // Check if the cigar contains indels
+        for (int i = 0; i < numCigarElements; i++) {
+            final CigarOperator e = read.getCigarElement(i).getOperator();
+            if (e == CigarOperator.INSERTION || e == CigarOperator.DELETION) {
+                sawIndel = true;
+                break;
+            }
+        }
+        if (!sawIndel) {
+            return new Tuple<>(bases, baseQualities);
         }
         else {
-            final byte[] paddedBases = new byte[CigarUtils.countRefBasesIncludingSoftClips(read, 0, cigar.numCigarElements())];
+            int numberRefBasesIncludingSoftclips = CigarUtils.countRefBasesIncludingSoftClips(read, 0, numCigarElements);
+            final byte[] paddedBases = new byte[numberRefBasesIncludingSoftclips];
+            final byte[] paddedBaseQualities = new byte[numberRefBasesIncludingSoftclips];
             int literalPos = 0;
             int paddedPos = 0;
-            for ( int i = 0; i < cigar.numCigarElements(); i++ ) {
-                final CigarElement ce = cigar.getCigarElement(i);
+            for ( int i = 0; i < read.numCigarElements(); i++ ) {
```

Comment:
Use the already-initialized `numCigarElements` here instead of `read.numCigarElements()`

Reply:
good catch!